src/tpm2_wrap.c

```diff
@@ -172,6 +172,272 @@ int wolfTPM2_Init(WOLFTPM2_DEV* dev, TPM2HalIoCb ioCb, void* userCtx)
+#endif
+        return BUFFER_E;
+    }
+
+    if (bufferSz < sizeof(key->pub.size) + sizeof(UINT16) + key->pub.size +
+                   sizeof(UINT16) + key->priv.size) {
+        return BUFFER_E;
+    }
+
+    /* Write size marker for the public part */
+    XMEMCPY(buffer + sz, &key->pub.size, sizeof(key->pub.size));
+    sz += sizeof(key->pub.size);
+
+    /* Write the public part with bytes aligned */
+    XMEMCPY(buffer + sz, pubAreaBuffer, sizeof(UINT16) + key->pub.size);
+    sz += sizeof(UINT16) + key->pub.size;
+
+    /* Write the private part, size marker is included */
+    XMEMCPY(buffer + sz, &key->priv, sizeof(UINT16) + key->priv.size);
+    sz += sizeof(UINT16) + key->priv.size;
+
+#ifdef WOLFTPM_DEBUG_VERBOSE
+    TPM2_PrintBin(buffer, sz);
+    printf("Getting %d bytes\n", (int)sz);
+#endif
+
+    return sz;
+}
+
+int wolfTPM2_SetKeyBlobFromBuffer(WOLFTPM2_KEYBLOB* key, byte *buffer,
+                                  word32 bufferSz)
+{
+    int rc = 0;
+    byte pubAreaBuffer[sizeof(TPM2B_PUBLIC)];
+    int pubAreaSize;
+    byte *runner = buffer;
+    size_t done_reading = 0;
+
+    if ((key == NULL) || (buffer == NULL) || (bufferSz <= 0)) {
+        return BAD_FUNC_ARG;
+    }
+
+    XMEMSET(key, 0, sizeof(WOLFTPM2_KEYBLOB));
+
+#ifdef WOLFTPM_DEBUG_VERBOSE
+    TPM2_PrintBin(buffer, bufferSz);
+    printf("Setting %d bytes\n", (int)bufferSz);
+#endif
+
+    if (bufferSz < done_reading + sizeof(key->pub.size)) {
+#ifdef WOLFTPM_DEBUG_VERBOSE
+        printf("Buffer size check failed (%d)\n",  bufferSz);
+#endif
+        return BUFFER_E;
+    }
+
+    XMEMCPY(&key->pub.size, runner, sizeof(key->pub.size));
+    runner += sizeof(key->pub.size);
+    done_reading += sizeof(key->pub.size);
+
+    if (bufferSz < done_reading + sizeof(UINT16) + key->pub.size) {
+#ifdef WOLFTPM_DEBUG_VERBOSE
+        printf("Buffer size check failed (%d)\n",  bufferSz);
+#endif
+        return BUFFER_E;
+    }
+
+    XMEMCPY(pubAreaBuffer, runner, sizeof(UINT16) + key->pub.size);
+    runner += sizeof(UINT16) + key->pub.size;
+    done_reading += sizeof(UINT16) + key->pub.size;
+
+    /* Decode the byte stream into a publicArea structure ready for use */
+    rc = TPM2_ParsePublic(&key->pub, pubAreaBuffer,
+        (word32)sizeof(pubAreaBuffer), &pubAreaSize);
+    if (rc != TPM_RC_SUCCESS) {
+        return rc;
+    }
+
+    if (bufferSz < done_reading + sizeof(key->priv.size)) {
+#ifdef WOLFTPM_DEBUG_VERBOSE
+        printf("Buffer size check failed (%d)\n",  bufferSz);
+#endif
+        return BUFFER_E;
+    }
+
+    XMEMCPY(&key->priv.size, runner, sizeof(key->priv.size));
+    runner += sizeof(key->priv.size);
+    done_reading += sizeof(key->priv.size);
+
+    if (bufferSz < done_reading + key->priv.size) {
+#ifdef WOLFTPM_DEBUG_VERBOSE
+        printf("Buffer size check failed (%d)\n",  bufferSz);
+#endif
+        return BUFFER_E;
+    }
+
+    XMEMCPY(key->priv.buffer, runner, key->priv.size);
+    runner += key->priv.size;
```

Comment:
These last two lines are dead code. 
```
runner += key->priv.size;
done_reading += key->priv.size;
```

Reply:
Yes.  And the optimizer will remove this from the executable, but this future proofs against someone adding more stuff and forgetting to add the incrementations.